<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="use_camera" default="False" />
    <arg name="image_width" default="1280" /> 
    <arg name="image_height" default="720" />
    <arg name="enable_blur" default="True" />
    <arg name="camera" default="/camera/color/image_raw" />
    <arg name="tf_prefix" default="marker_id" />
    <arg name="marker_size" default="0.9" doc="length of the physical marker's side in meters"/>
    <arg name="marker_fx" default="956.897217"/>
    <arg name="marker_fy" default="988.634277"/>
    <arg name="marker_cx" default="640.51"/>
    <arg name="marker_cy" default="360.25"/>
    <arg name="blur_window_size" default="7" />

   <!--   usb_cam node -->
  <group if="$(arg use_camera)" >
    <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen">
      <param name="video_device" value="/dev/video1" />
      <param name="image_width" value="$(arg image_width)" />
      <param name="image_height" value="$(arg image_height)" />
      <param name="pixel_format" value="yuyv" />
      <param name="camera_frame_id" value="usb_cam" />
      <param name="io_method" value="mmap"/>
    </node>
  </group>

  <!-- 1920x1080  -->
  <node pkg="apriltag_detector" type="node" name="apriltag_detect_node" output="screen">
      <param name="camera" type="string" value="$(arg camera)"/>
      <param name="tf_prefix" type="string" value="$(arg tf_prefix)"/>
      <param name="marker_size" type="double" value="$(arg marker_size)"/>
      <param name="marker_fx" type="double" value="$(arg marker_fx)"/>
      <param name="marker_fy" type="double" value="$(arg marker_fy)"/>
      <param name="marker_cx" type="double" value="$(arg marker_cx)"/>
      <param name="marker_cy" type="double" value="$(arg marker_cy)"/>
      <param name="image_width" type="double" value="$(arg image_width)"/>
      <param name="image_height" type="double" value="$(arg image_height)"/>
      <param name="enable_blur" type="bool" value="$(arg enable_blur)"/>
      <param name="blur_window_size" type="double" value="$(arg blur_window_size)"/>
  </node>
</launch>
